cmd/handler-utils.go

```diff
@@ -65,6 +65,31 @@ var supportedHeaders = []string{
 	// Add more supported headers here.
 }
 
+// Checks if metadata list in the request has updated by
+// verifying source metadata list.
+func isMetadataUpdate(srcMetadata, newMetadata map[string]string) bool {
+	if len(newMetadata) == 0 {
+		return false
+	}
+	for nKey, nVal := range newMetadata {
+		sVal, ok := srcMetadata[nKey]
+		if !ok {
+			return true
+		}
+		if sVal != nVal {
+			return true
+		}
+	}
+	return false
+}
+
+// merge new metadata with existing source metadata.
+func mergeNewMetadata(srcMetadata, newMetadata map[string]string) {
```

Comment:
`func mergeMetadata(metadata ...map[string]string) map[string]string`

or

`func mergeMetadata(metadata1, metadata2 map[string]string) map[string]string`

will be better functions